Xamarin.Forms.Platform.iOS/Renderers/LabelRenderer.cs

```diff
@@ -174,6 +203,12 @@ protected override void OnElementPropertyChanged(object sender, PropertyChangedE
 				UpdateMaxLines();
 		}
 
+		void ElementPropertyChanging(object sender, PropertyChangingEventArgs e)
+		{
+			if (s_perfectSizeSet.Contains(e.PropertyName))
```

Comment:
Could we get rid of all the *Changing* code in this pr and just move this statement up to

https://github.com/xamarin/Xamarin.Forms/pull/4448/files#diff-381c6c37e826f38028e8841495e4f78eR176

I tested it a little bit and it seems to work. Is there a scenario where this code needs to be part of *Changing* opposed to at the top of *Changed*  ?

Reply:
That will be a little different from what it was before. I could break back the Changing. but in my head makes sense to invalidate before so we make sure  when any re layout trigged on PropertyChanged the flag is false

Reply:
> That will be a little different from what it was before.

In what way?

Reply:
https://github.com/xamarin/Xamarin.Forms/pull/4448/files#diff-381c6c37e826f38028e8841495e4f78eL256

is inside some methods it's not invalidated always.

Reply:
Can you explain why setting it on `Changing` is different from/better than setting it on `Changed`? We can still use the HashSet to limit which properties invalidates, right?

Reply:
Right we can still use. But imagine other parts of the Native Layout on the platform are also hooked to Changed events, and they will trigger the native `GetDesiredSize` to fire, this will try prevent that case.